### ISSUE TITLE
Request for [D,R] RIFE trained on Style loss, also called Gram matrix loss (the best perceptual loss function)

### ISSUE THREAD

**Issue by AIVFI** [zzh-tech/InterpAny-Clearer#12]
You were right to write [here](https://github.com/zzh-tech/BiT/issues/5#issuecomment-1597740948) that your new work may be of great interest to me. **The level of detail retention is really impressive** when compared to the baseline models and **your work could revolutionise the way VFI models are developed**, especially those for practical applications.

**I am particularly grateful to you for training the [D,R] RIFE-vgg model and the comparison of x128 interpolation with the [D,R] RIFE and [T] RIFE models.** You recommend [D,R] RIFE for more stable results and after seeing the GIF files with the interpolation results I fully share your opinion. These results mean that I will have to completely revise my introduction to [Video Frame Interpolation Rankings and Video Deblurring Rankings](https://github.com/AIVFI/Video-Frame-Interpolation-Rankings-and-Video-Deblurring-Rankings).

I think, however, that **there is a solution** to get all the advantages of the [D,R] RIFE-vgg and [D,R] RIFE models while eliminating their disadvantages. That solution is **Style loss**, also called **Gram matrix loss**.

The first time Style loss to train the video frame interpolation model was used by **Google Research** when training their FILM-𝓛<sub>*S*</sub> model, link: [![ECCV](https://img.shields.io/badge/2022-ECCV-67cd84)](https://www.ecva.net/papers/eccv_2022/papers_ECCV/html/4614_ECCV_2022_paper.php) The second time this loss was used by **Disney Research** when training their UGFI 𝓛<sub>*S*</sub> model, link: [![CVPR](https://img.shields.io/badge/2023-CVPR-1e407f)](https://openaccess.thecvf.com/content/CVPR2023/html/Plack_Frame_Interpolation_Transformer_and_Uncertainty_Guidance_CVPR_2023_paper.html) 

Both models achieved some of the best and maybe even the best LPIPS results (because we do not have a direct comparison with the other two models of the top four, and the results are very close to each other):

**Vimeo-90K triplet: LPIPS<=0.017 [excluding LPIPS(SqueezeNet) results]**
| Rank | &nbsp;&nbsp;&nbsp;&nbsp;Model&nbsp;&nbsp;&nbsp;&nbsp; | &nbsp;&nbsp;&nbsp;LPIPS&nbsp;↓&nbsp;&nbsp;&nbsp;<br />{Input&nbsp;fr.} | Training<br />dataset | Official<br />&nbsp;&nbsp;repository&nbsp;&nbsp; | Practical<br />model | VapourSynth |
|:---:|:---:|:---:|:---:|:---:|:---:|:---:|
| 1 | EAFI-𝓛<sub>*ecp*</sub><br />[![arXiv](https://img.shields.io/badge/2022-arXiv-b31b1b)](https://arxiv.org/abs/2207.12305) |  **0.012** {2}<br />[![arXiv](https://img.shields.io/badge/2022-arXiv-b31b1b)](https://arxiv.org/abs/2207.12305) | Vimeo-90K triplet | - | EAFI-𝓛<sub>*ecp*</sub> | - |
| 2 | UGFI 𝓛<sub>*S*</sub><br />[![CVPR](https://img.shields.io/badge/2023-CVPR-1e407f)](https://openaccess.thecvf.com/content/CVPR2023/html/Plack_Frame_Interpolation_Transformer_and_Uncertainty_Guidance_CVPR_2023_paper.html) | **0.0126** {2}<br />[![CVPR](https://img.shields.io/badge/2023-CVPR-1e407f)](https://openaccess.thecvf.com/content/CVPR2023/html/Plack_Frame_Interpolation_Transformer_and_Uncertainty_Guidance_CVPR_2023_paper.html) | Vimeo-90K triplet | - | UGFI 𝓛<sub>*S*</sub> | - |
| 3 | SoftSplat - 𝓛<sub>*F*</sub><br />[![CVPR](https://img.shields.io/badge/2020-CVPR-1e407f)](https://openaccess.thecvf.com/content_CVPR_2020/html/Niklaus_Softmax_Splatting_for_Video_Frame_Interpolation_CVPR_2020_paper.html) |  **0.013** {2}<br />[![CVPR](https://img.shields.io/badge/2020-CVPR-1e407f)](https://openaccess.thecvf.com/content_CVPR_2020/html/Niklaus_Softmax_Splatting_for_Video_Frame_Interpolation_CVPR_2020_paper.html) | Vimeo-90K triplet | [![GitHub Stars](https://img.shields.io/github/stars/sniklaus/softmax-splatting?logo=GitHub&label=Stars)](https://github.com/sniklaus/softmax-splatting) | SoftSplat - 𝓛<sub>*F*</sub> | - |
| 4 | FILM-𝓛<sub>*S*</sub><br />[![ECCV](https://img.shields.io/badge/2022-ECCV-67cd84)](https://www.ecva.net/papers/eccv_2022/papers_ECCV/html/4614_ECCV_2022_paper.php) | **0.0132** {2}<br />[![CVPR](https://img.shields.io/badge/2023-CVPR-1e407f)](https://openaccess.thecvf.com/content/CVPR2023/html/Plack_Frame_Interpolation_Transformer_and_Uncertainty_Guidance_CVPR_2023_paper.html) | Vimeo-90K triplet | [![GitHub Stars](https://img.shields.io/github/stars/google-research/frame-interpolation?logo=GitHub&label=Stars)](https://github.com/google-research/frame-interpolation) | [FILM-𝓛<sub>*S*</sub>](https://github.com/google-research/frame-interpolation#pre-trained-models) | - |
| 5 | EDSC_s-𝓛<sub>*F*</sub><br />[![TPAMI](https://img.shields.io/badge/2021-TPAMI-fefd02)](https://ieeexplore.ieee.org/document/9501506) |  **0.016** {2}<br />[![arXiv](https://img.shields.io/badge/2020-arXiv-b31b1b)](https://arxiv.org/abs/2006.08070) | Vimeo-90K triplet | [![GitHub Stars](https://img.shields.io/github/stars/Xianhang/EDSC-pytorch?logo=GitHub&label=Stars)](https://github.com/Xianhang/EDSC-pytorch) | [EDSC_s-𝓛<sub>*F*</sub>](https://github.com/Xianhang/EDSC-pytorch#pre-trained-models) | - |
| 6 | CtxSyn - 𝓛<sub>*F*</sub><br />[![CVPR](https://img.shields.io/badge/2018-CVPR-1e407f)](https://openaccess.thecvf.com/content_cvpr_2018/html/Niklaus_Context-Aware_Synthesis_for_CVPR_2018_paper.html) | **0.017** {2}<br />[![CVPR](https://img.shields.io/badge/2020-CVPR-1e407f)](https://openaccess.thecvf.com/content_CVPR_2020/html/Niklaus_Softmax_Splatting_for_Video_Frame_Interpolation_CVPR_2020_paper.html) | proprietary | - | CtxSyn - 𝓛<sub>*F*</sub> | - |

However, the most interesting thing is the visual comparison of the three loss functions. In my opinion, Style loss clearly gives the best result perceptually:

![FILM - Loss Functions Ablation](https://film-net.github.io/static/images/loss_ablation.png)
Source: FILM - Loss Functions Ablation https://film-net.github.io/

Furthermore, **interpolation with FILM-𝓛<sub>*S*</sub> can eliminate the artefacts seen with FILM-𝓛<sub>1</sub>**, as can be seen in [Fig. 1](https://www.ecva.net/papers/eccv_2022/papers_ECCV/papers/136670244-supp.pdf) in Supplementary Material.

More details on Style loss and more examples are on YouTube: https://www.youtube.com/watch?v=OAD-BieIjH4&t=160s 

UGFI model trained with Style loss also retains an amazing amount of fine detail, as the examples at the bottom of [Figure 6](https://openaccess.thecvf.com/content/CVPR2023/supplemental/Plack_Frame_Interpolation_Transformer_CVPR_2023_supplemental.pdf) in Supplementary Material show particularly well.

The Style loss equation can be found in [Sec.3.1](https://www.ecva.net/papers/eccv_2022/papers_ECCV/papers/136670244.pdf)  
The loss combination weights for the FILM-𝓛<sub>*S*</sub> model are in [Sec. 1.1](https://www.ecva.net/papers/eccv_2022/papers_ECCV/papers/136670244-supp.pdf) in Supplementary Material.  
The loss combination weights for the UGFI 𝓛<sub>*S*</sub> model are in [Sec. 3.3](https://openaccess.thecvf.com/content/CVPR2023/papers/Plack_Frame_Interpolation_Transformer_and_Uncertainty_Guidance_CVPR_2023_paper.pdf)

You did a great job with the [T] RIFE, [D,R] RIFE, [D,R] RIFE-vgg comparison. You've shown something I haven't seen anywhere so far, which is that at x128 interpolation VGG Loss can give more messy artifacts than the benefit of preserving fine detail.  I think now more people may be seriously considering whether to use the perceptual loss function for practical purposes.

Therefore, **I have a great request to you to train the [D,R] RIFE model using Style loss and compare it with the other 3 models** whose x128 interpolation results you showed as GIF files. **This may be the best model for practical use** - retaining the most detail with your method combined with training using Style loss without the creation of messy artefacts.

I would like to include the results of the comparison you manage to achieve in the introduction to [Video Frame Interpolation Rankings and Video Deblurring Rankings](https://github.com/AIVFI/Video-Frame-Interpolation-Rankings-and-Video-Deblurring-Rankings) to draw the attention of other researchers to how to train VFI models for practical applications and of course to your method as well. 

Also have a look at what your neighbours in your city developed a month ago: a **no-reference** [Perceptual Quality Assessment for Video Frame Interpolation](https://arxiv.org/abs/2312.15659), in particular TABLE I.

**Comment by zzh-tech** [comment]
Hi, thanks for the advice and sorry for the late reply.
I can try it when I'm not busy.
But if you or anyone can help to run this experiment, that would be a big help.

**Comment by AIVFI** [comment]
Many thanks for your response and willingness to solve this intriguing problem.

I completely understand the lack of time as I know this problem from my example. I am ashamed of how my rankings look and the fact that I don't have time to update them. Unfortunately all my activity here is my hobby and my job for life is something completely different.

I would love to help, but unfortunately I'm not even a programmer, let alone something as complicated as training AI models. I'm planning to buy an NVIDIA GeForce RTX 5090 graphics card early next year, and if someone develop the software with a GUI to train AI models, then of course I'd be happy to help with training or other testing. 

I will, however, try to help as much as I can to publicise this thread and maybe someone else will get interested in the topic and help with this experiment.

I think one thing would help a lot to increase interest in this experiment and your InterpAny-Clearer method in general.

If you could add to the comparison:

[T] RIFE
[D,R] RIFE (Ours)
[D,R] RIFE-vgg (Ours)

one more model, without any training:

**[T] RIFE v4.15**
link: https://github.com/hzwer/Practical-RIFE#model-list 

I think generating an animated GIF file with x128 interpolation won't take much time and will greatly increase interest in your method and allow me to better publicise InterpAny-Clearer among the many enthusiasts who use this very practical model on a daily basis, rather than the base version of RIFE. 

In my opinion, it is the comparison with the practical, current best version of RIFE i.e. RIFE v4.15 that will be the best confirmation of the need for the InterpAny-Clearer and give me the needed proof to publicise in my rankings and beyond the need to use this improvement in all practical models in the future, not only with RIFE.

Of course, I would also like to know what loss function to recommend in the introduction to my rankings so that future models use it too. This is why it is so important to find out which loss function works best with InterpAny-Clearer. You recommend using [D,R] RIFE rather than [D,R] RIFE-vgg and I agree that the example given with the young woman's face justifies this. However, I think it is worth doing this experiment to see if Style loss, also called Gram matrix loss will give an even better effect. Then we will have a clear proof of which loss function InterpAny-Clearer works best with.

**Comment by AIVFI** [comment]
see RIFE v4.17 - trained on Style loss, also called Gram matrix: https://github.com/hzwer/Practical-RIFE#model-list
